app/javascript/gobierto_dashboards/modules/contracts_controller.js

```diff
@@ -137,23 +146,25 @@ export class ContractsController {
       contract.final_amount = final_amount;
       contract.initial_amount = initial_amount;
       contract.range = rangeFormat(+final_amount);
+      contract.start_date_year = contract.start_date != undefined && contract.start_date != '' ? (new Date(contract.start_date).getFullYear()) : contract.start_date;
```

Comment:
To check for `undefined` and length, you can leave it as:

```suggestion
      contract.start_date_year = contract.start_date ? (new Date(contract.start_date).getFullYear()) : contract.start_date;
```
Undefineds, nulls, and empty lengths are evaluated to `false`

Ay, ay, ay, that destructuring...